src/coreclr/vm/loaderallocator.cpp

```diff
@@ -550,10 +570,13 @@ void LoaderAllocator::GCLoaderAllocators(LoaderAllocator* pOriginalLoaderAllocat
         pDomainLoaderAllocatorDestroyIterator = pDomainLoaderAllocatorDestroyIterator->m_pLoaderAllocatorDestroyNext;
     }
 
-    // If the original LoaderAllocator was not processed, it is most likely a LoaderAllocator without any loaded DomainAssembly
+    // If the original LoaderAllocator was not processed, it is a LoaderAllocator without any loaded DomainAssembly
     // But we still want to collect it so we add it to the list of LoaderAllocator to destroy
-    if (!isOriginalLoaderAllocatorFound && !pOriginalLoaderAllocator->IsAlive())
+    if (!isOriginalLoaderAllocatorFound && !pOriginalLoaderAllocator->Id()->HasAttachedDynamicAssemblies() && !pOriginalLoaderAllocator->IsAlive())
```

Comment:
When does the original loader allocator get destroyed when it has some attached dynamic assemblies?

Reply:
Correct, we don't re-run the LoaderAllocator::Destroy, but we will build a list of which loader allocators to really clean up,  so we don't need to actually re-run it. The critical parts end up getting run later through the GC process. What parts of LoaderAllocator::Destroy are you interested in?